### DISCUSSION TITLE
V9.5.0, how to collect metrics from prometheus to skywalking oap

### DISCUSSION THREAD

**Discussion by jiahaopei** [apache/skywalking#11275] [Q&A]
i use otel-collector collect node-exporter metrics, and send to skywalking. But node-exporter metrics and other metrics are already in prometheus, so i want to send prometheus data directly to skywalking.
but first: i don't know if there is a way to use otel-collector collect prometheus?
second: is there a way skywalking can directly collect prometheus metrics?

**Comment by wu-sheng** [comment]
Do you ever read any document?

**Reply by jiahaopei** [reply]
i have read some document. such as https://skywalking.apache.org/docs/main/v9.5.0/en/setup/backend/opentelemetry-receiver/.

**Reply by jiahaopei** [reply]
Through these document, i realize MicroMeter Observation and self Observation ,but i don't find what i ask.

**Reply by wu-sheng** [reply]
What do you mean? You were asking otel? Now changed?

**Reply by jiahaopei** [reply]
i want to send prometheus metrics data to skywalking. is otel can do this?  Or skywalking now can directly receive prometheus data without otel?

**Reply by wu-sheng** [reply]
> i have read some document. such as https://skywalking.apache.org/docs/main/v9.5.0/en/setup/backend/opentelemetry-receiver/.

Doesn't this provide a clear answer?

**Reply by jiahaopei** [reply]
i can realize this: Prometheus node-exporter(VMs) -> OpenTelemetry Collector – OTLP exporter –> SkyWalking OAP Server. But Prometheus -> OpenTelemetry Collector – OTLP exporter –> SkyWalking OAP Server, i guess it may need something to export prometheus data to otle, but i can't find one way

**Reply by wu-sheng** [reply]
Do you know what is Prometheus endpoint? It is OTEL fetching from it.

**Reply by jiahaopei** [reply]
Sorry, i didn't express it clearly. Prometheus endpoint export itself metrics. What i want is Prometheus collected metrics, such as node exporter already export vm metrics to prometheus, so i don't want to export it to otel again, i want export what Prometheus has collected to otel.

**Reply by wu-sheng** [reply]
First of all, this is knowledge of Prometheus. Why do you think we should answer that? Your asking doesn't make sense.

**Reply by jiahaopei** [reply]
I don't know if skywalking support directly from prometheus. Such as Prometheus remote write to skywalking oap

**Reply by wu-sheng** [reply]
No, we are not a database of Prometheus, of course we can't.

**Reply by jiahaopei** [reply]
OH! Whether Skywalking kafka fetcher support prometheus metrics? If not, Prometheus remote write to kafka, otel receive from kafka and send to skywalking, i think it may work

**Reply by wu-sheng** [reply]
I really can't get what you are *** talking about. I am done here. The solution is there so simply, and you are just adding random things.

**Reply by jiahaopei** [reply]
Through all discuss, i just want find way to realize sending what Prometheus has collcted to skywalking. Maybe i am stupid, but you don't need to use ***.

**Reply by wu-sheng** [reply]
You repeatedly skipped my answer, a capability of OTEL already provided, but kept asking Prometheus things. 
How to use Prometheus and OTEL is your responsibility. That is your deployment.

**Reply by jiahaopei** [reply]
Your answer is always a question. I think if you say this at first, maybe i would not keep asking. At last, still thanks for your reply
